internal/resources/metal/vlan/models.go

```diff
@@ -21,29 +21,34 @@ type DataSourceModel struct {
 	AssignedDevicesIds types.List   `tfsdk:"assigned_devices_ids"`
 }
 
-func (m *DataSourceModel) parse(vlan *packngo.VirtualNetwork) (d diag.Diagnostics) {
-	m.ID = types.StringValue(vlan.ID)
-	m.VlanID = types.StringValue(vlan.ID)
-	m.Vxlan = types.Int64Value(int64(vlan.VXLAN))
+func (m *DataSourceModel) parse(vlan *metalv1.VirtualNetwork) (d diag.Diagnostics) {
+	m.ID = types.StringValue(vlan.GetId())
+	m.VlanID = types.StringValue(vlan.GetId())
+	m.Vxlan = types.Int64Value(int64(vlan.GetVxlan()))
 	m.Facility = types.StringValue("")
 
-	if vlan.Description != "" {
-		m.Description = types.StringValue(vlan.Description)
+	if vlan.GetDescription() != "" {
+		m.Description = types.StringValue(vlan.GetDescription())
```

Comment:
Empty or not, when would we not want the model description to match the upstream description?

Reply:
https://github.com/equinix/terraform-provider-equinix/pull/642

Reply:
There may be other ways to address the bug fixed by that previous PR; I forget what else, if anything, I tried at that time, but I think I was aiming to maintain backwards compatibility.  The effect of this if block is that, if the description is null or `""` in the response, it is null in the model.  I don't recall if the API meaningfully differentiates between a null VLAN description and an empty VLAN description.